imageprocessor/imageprocessor-core/src/test/java/pl/edu/uj/kimage/processingFlow/ImageContainerTest.java

```diff
@@ -1,55 +1,58 @@
+        int width = 2, height = 2;
+        Color[] colors = new Color[]{new Color(255, 0, 0), new Color(0, 255, 0),
+                                     new Color(0, 0, 255), new Color(255, 0, 0)};
+
+        Image image = new Image(width, height, colors);
+        assertThat(image.getWidth()).isEqualTo(width);
+        assertThat(image.getHeight()).isEqualTo(height);
     }
 
     @Test
-    public void oneChannelOperation() {
-        Image image = new Image(256, 256);
-        int red = 128;
-        image.setRed(5, 5, red);
-        assertThat(image.getRed(5, 5)).isEqualTo(red);
+    public void channelColorAccessorsReturnValidValues() {
+        int width = 1, height = 1;
+        int redChannel = 255, greenChannel = 0, blueChannel = 0;
+        Color[] colors = new Color[]{new Color(redChannel, greenChannel, blueChannel)};
+
+        Image image = new Image(width, height, colors);
+        assertThat(image.getRed(0, 0)).isEqualTo(redChannel);
+        assertThat(image.getGreen(0, 0)).isEqualTo(greenChannel);
+        assertThat(image.getBlue(0, 0)).isEqualTo(blueChannel);
     }
 
     @Test
-    public void multiChannelOperation() {
-        Image image = new Image(256, 256);
-
-        int x = 0, y = 0;
-        int red = 0, green = 128, blue = 200;
-
-        image.setRGB(x, y, red, green, blue);
-
-        int[] colors = image.getRGB(x, y);
-
-        assertThat(colors[0]).isEqualTo(red);
-        assertThat(colors[1]).isEqualTo(green);
-        assertThat(colors[2]).isEqualTo(blue);
+    public void channelColorAccessorsReturnClampedValues() {
+        int width = 1, height = 1;
+        int redChannel = 0, greenChannel = -25, blueChannel = 280;
+        Color[] colors = new Color[]{new Color(redChannel, greenChannel, blueChannel)};
+
+        Image image = new Image(width, height, colors);
+        assertThat(image.getRed(0, 0)).isEqualTo(0);
+        assertThat(image.getGreen(0, 0)).isEqualTo(0);
+        assertThat(image.getBlue(0, 0)).isEqualTo(255);
     }
 
     @Test
-    public void colorValuesClamping() {
-        Image image = new Image(256, 256);
-
-        int x = 0, y = 0;
-        int red = 0, green = -5, blue = 300;
-
-        image.setRGB(0, 0, red, green, blue);
-
-        int[] colors = image.getRGB(x, y);
-
-        assertThat(colors[0]).isEqualTo(0);
-        assertThat(colors[1]).isEqualTo(0);
-        assertThat(colors[2]).isEqualTo(255);
+    public void fullColorAccessorsReturnValidData() {
+        int width = 2, height = 2;
+        Color[] colors = new Color[]{new Color(255, 0, 0), new Color(0, 255, 0),
+                                     new Color(0, 0, 255), new Color(255, 0, 255)};
+
+        Image image = new Image(width, height, colors);
+        Color rgba = image.getRGBA(1, 1);
+        assertThat(rgba.getRed()).isEqualTo(255);
+        assertThat(rgba.getGreen()).isEqualTo(0);
+        assertThat(rgba.getBlue()).isEqualTo(255);
+        assertThat(rgba.getAlpha()).isEqualTo(255);
```

Comment:
where is the test for
`throw new IllegalArgumentException("Size of image does not match with input color data size.");`?